### ISSUE TITLE
Handle documents that are malformed or don't have an SVG element

### ISSUE THREAD

**Issue by jamesmartin** [jamesmartin/inline_svg#60]
Currently a pattern in the transform classes (E.g. [`ClassAttribute`](https://github.com/jamesmartin/inline_svg/blob/33b2fcd62ecc3b068e289153ae4c7dc015466ad5/lib/inline_svg/transform_pipeline/transformations/class_attribute.rb#L4-L5) is to just assume that the document they're given contains a root-level SVG element:

```ruby
svg = doc.at_css "svg"
#... do some stuff assuming SVG returned something
```

When a document is malformed this causes deep errors inside rails apps that are hard to track down, like this:

```ruby
undefined method `[]' for nil:NilClass
```

Ideally we should have a way for all transforms to get hold of the SVG part of the document, if one exists, and bail cleanly if not.

Here's an example of an XML document that *looks* OK, but will parse strangely and end up with `nil` for the SVG part (notice how the `<title>` element is outside of the `<svg>` document):

```xml
<?xml version="1.0" encoding="utf-8"?>
<title>Some title</title>
<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 40 40" version="1.1">
	<g stroke="none" stroke-width="1" fill="none" fill-rule="evenodd"><g fill="#ffffff"><path d="M20 40C31 40 40 31 40 20 40 9 31 0 20 0 9 0 0 9 0 20 0 31 9 40 20 40L20 40ZM20 38C10.1 38 2 29.9 2 20 2 10.1 10.1 2 20 2 29.9 2 38 10.1 38 20 38 29.9 29.9 38 20 38L20 38ZM29.3 20L15 28.8 15 11.1 29.3 20 29.3 20Z"></path></g></g>
</svg>
```

**Comment by Geetakokate** [comment]
I got same error, undefined method `[]' for nil:NilClass.

**Comment by jamesmartin** [comment]
> I got same error, undefined method `[]' for nil:NilClass.

@Geetakokate thanks for the feedback. Until this bug is fixed, you can workaround the problem by only using SVG documents that contain a `<svg>` element at the root. E.g.

Bad :-1: :

```xml
<foo>something</foo>
<svg>...</svg>
```

Good 👍 : 

```xml
<svg>
  <foo>something</foo>
</svg>
```

**Comment by jamesmartin** [comment]
Resolved by #65. Will be released in v1.2.2.
